pkg/controller/controller_test.go

```diff
@@ -976,66 +978,7 @@ func newTestController(t *testing.T) (
 		t.Fatal(err)
 	}
 
-	return fakeKubeClient, fakeCatalogClient, fakeBrokerClient, testController.(*controller), serviceCatalogSharedInformers
-}
-
-type testControllerWithBrokerServer struct {
-	FakeKubeClient      *clientgofake.Clientset
-	FakeCatalogClient   *servicecatalogclientset.Clientset
-	Controller          *controller
-	Informers           v1alpha1informers.Interface
-	BrokerServerHandler *fakebrokerserver.Handler
-	BrokerServer        *httptest.Server
-}
-
-func (t *testControllerWithBrokerServer) Close() {
-	t.BrokerServer.Close()
-}
-
-func newTestControllerWithBrokerServer(
-	brokerUsername,
-	brokerPassword string,
-) (*testControllerWithBrokerServer, error) {
-	// create a fake kube client
-	fakeKubeClient := &clientgofake.Clientset{}
-	// create a fake sc client
-	fakeCatalogClient := &servicecatalogclientset.Clientset{}
-
-	brokerHandler := fakebrokerserver.NewHandler()
-	brokerServer := fakebrokerserver.Run(brokerHandler, brokerUsername, brokerPassword)
-	brokerClFunc := fakebrokerserver.NewCreateFunc(brokerServer, brokerUsername, brokerPassword)
-
-	// create informers
-	informerFactory := servicecataloginformers.NewSharedInformerFactory(fakeCatalogClient, 0)
-	serviceCatalogSharedInformers := informerFactory.Servicecatalog().V1alpha1()
-
-	fakeRecorder := record.NewFakeRecorder(5)
-
-	// create a test controller
-	testController, err := NewController(
-		fakeKubeClient,
-		fakeCatalogClient.ServicecatalogV1alpha1(),
-		serviceCatalogSharedInformers.Brokers(),
-		serviceCatalogSharedInformers.ServiceClasses(),
-		serviceCatalogSharedInformers.Instances(),
-		serviceCatalogSharedInformers.Bindings(),
-		brokerClFunc,
-		24*time.Hour,
-		true, /* enable OSB context profile */
-		fakeRecorder,
-	)
-	if err != nil {
-		return nil, err
-	}
-
-	return &testControllerWithBrokerServer{
-		FakeKubeClient:      fakeKubeClient,
-		FakeCatalogClient:   fakeCatalogClient,
-		Controller:          testController.(*controller),
-		Informers:           serviceCatalogSharedInformers,
-		BrokerServerHandler: brokerHandler,
-		BrokerServer:        brokerServer,
-	}, nil
```

Comment:
why remove this? I think you changed the single test that relied on it, but we have plans to use the fake broker server in the future, in integration tests and select unit tests (like in https://github.com/kubernetes-incubator/service-catalog/pull/923)